commands/bootstrap/index.js

```diff
@@ -66,6 +74,7 @@ class BootstrapCommand extends Command {
       npmClient,
       npmClientArgs,
       mutex,
+      npmCiMode,
```

Comment:
This should be removed, in favor of conditional assignment of `this.npmConfig.subCommand`:

```js
if (npmClient === "npm" && this.options.ci && hasNpmCI()) {
  this.npmConfig.subCommand = "ci";
}
```

I sketched out the implementation of `hasNpmCI()` [here](https://runkit.com/embed/auv7hpnsx7f4), where you can see test examples too.

`commands/bootstrap/lib/make-npm-feature-predicate.js`:
```js
"use strict";

const semver = require("semver");

module.exports = makeNpmFeaturePredicate;

function makeNpmFeaturePredicate(range) {
  return () => Boolean(semver.satisfies(getNpmVersion(), range));
}

function getNpmVersion() {
  const userAgent = process.env.npm_config_user_agent;

  return (/npm\/(\d+\.\d+\.\d+)/.exec(userAgent) || [""]).pop();
}
```

`commands/bootstrap/index.js`:
```js
const makeNpmFeaturePredicate = require("./lib/make-npm-feature-predicate");

// https://github.com/npm/npm/releases/tag/v5.7.0
const hasNpmCI = makeNpmFeaturePredicate(">=5.7.0");
```